packages/nestjs-query/src/dataProvider/index.ts

```diff
@@ -419,30 +419,37 @@ const dataProvider = (client: GraphQLClient): Required<DataProvider> => {
       return (client as any).url; // url field in GraphQLClient is private
     },
     custom: async ({ url, method, headers, meta }) => {
-      if (url) {
-        client.setEndpoint(url);
-      }
+      const SUPPORTED_METHODS = ["get", "post"];
+      const requestUrl = url || (client as any).url;
 
-      if (headers) {
-        client.setHeaders(headers);
+      if (!SUPPORTED_METHODS.some((it) => it === method)) {
+        throw Error(`GraphQL does not support ${method} method.`);
       }
 
+      const validMethod = method as "get" | "post";
+
+      const _client = new GraphQLClient(requestUrl, {
+        ...client.requestConfig,
+        method: validMethod,
+        headers,
```

Comment:
```suggestion
        headers: { ..client.requestConfig.headers, ...headers },
```

I think would be nice to not lose headers, if there are any.

Reply:
nice catch